addons/udes_stock_routing/models/stock_move.py

```diff
@@ -105,15 +105,147 @@ def _get_new_picking_values(self):
 
         return values
 
+    def _split_pick_for_rule(self, assigned_moves, rule):
+        """
+        Based on the current rule, and the assigned moves, split
+        the moves to a backorder, using the rule to determine the values of the new picking.
+
+        param: assigned_moves: stock.move(x,) recordset.
+        param: rule: stock.rule(x,) singleton.
+        """
+        StockMoveObj = self.env["stock.move"]
+        res = StockMoveObj.browse()
+        # Rip the moves to the operation type of the rule.
+        split_pick_vals = {
+            "location_id": rule.location_src_id.id,
+            "location_dest_id": rule.location_id.id,
+            "picking_type_id": rule.picking_type_id.id,
+            "origin": self.picking_id.origin,
+        }
+        # We only need to split the moves to a backorder if the picking type
+        # of the rule differs from the assigned moves picking type. If it
+        # is the same picking type, any post assign behaviour has already run, and
+        # the move lines and moves already exist on the picking of the correct type.
+        if assigned_moves.picking_type_id != rule.picking_type_id:
+            # TODO palabaster: This isn't ideal. Could there be cases where this is wrong?
+            # Use any picking from here (there could be multiple) to use as the basis for the backorder.
+            split_pick = assigned_moves.picking_id[0].split_move_lines_to_backorder(
```

Comment:
I think moves should be of same picking isnt it?
maybe above smth like 
assigned_moves.picking_id.ensure_one()
```suggestion
            split_pick = assigned_moves.picking_id.split_move_lines_to_backorder(
```

Reply:
no, refactoring can split moves to multiple pickings on assignment

Reply:
i have changed code in _action_assign now so assigned_moves won't ever include multiple pickings, as that is the source of some other pains, so this will be removed.